File: backend-ai/app/models/company.py
"""
회사 DB 모델
"""
from sqlalchemy import Column, Integer, String, Text, DateTime
from datetime import datetime


class Company:
    """
    회사 모델
    """
    __tablename__ = "companies"

    id = Column(Integer, primary_key=True)
    name = Column(String(255), nullable=False)  # 회사명
    description = Column(Text)  # 회사 설명
    website = Column(String(255))  # 회사 웹사이트
    industry = Column(String(100))  # 업종
    created_at = Column(DateTime, default=datetime.utcnow)  # 생성일
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)  # 수정일

    # TODO: 관계 설정

File: backend-core/src/main/java/com/skouter/recruitai/domain/company/entity/Company.java
package com.skouter.recruitai.domain.company.entity;

import com.skouter.recruitai.global.common.entity.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;

/**
 * 기업 엔티티
 * - 기업 정보 저장
 */
@Entity
@Table(name = "companies", indexes = {
        @Index(name = "idx_name", columnList = "name")
})
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Company extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    @Column
    private String website;

    @Column
    private String logoUrl;

    @Column
    private String industry;

    @Column
    private String scale;

    @Column
    private String location;

    /**
     * 기업 정보 업데이트
     *
     * @param name 기업명
     * @param description 설명
     * @param website 웹사이트
     * @param industry 산업
     * @param scale 규모
     * @param location 위치
     */
    public void updateCompanyInfo(String name, String description, String website,
                                   String industry, String scale, String location) {
        if (name != null) this.name = name;
        if (description != null) this.description = description;
        if (website != null) this.website = website;
        if (industry != null) this.industry = industry;
        if (scale != null) this.scale = scale;
        if (location != null) this.location = location;
    }
}
